### PR TITLE
deprecate some functions

### PR DESCRIPTION
**Pull request by minggo**
Re: https://github.com/cocos/cocos-engine/issues/13904

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/geometry/aabb.ts [hunk 1]
```diff
@@ -76,10 +77,18 @@ export class AABB {
       * 克隆一个 AABB，其会创建出一个值跟输入参数`a`一样的 AABB 实例。注意，每次调用 `clone` 都会创建出新实例，尽可能使用 `copy` 方法以减小 GC 压力。
       * @param a @zh 克隆的目标。 @en The target object to be cloned.
       * @returns @zh 克隆出的 AABB 实例。@en The cloned AABB instance.
+      * @deprecated since v3.8.4. Please use the corresponding instance method instead.
       */
     public static clone (a: AABB | Readonly<AABB>): AABB {
-        return new AABB(a.center.x, a.center.y, a.center.z,
-            a.halfExtents.x, a.halfExtents.y, a.halfExtents.z);
+        warn('The static method AABB.clone has been deprecated. Please use the corresponding instance method instead.');
```
Comment (dumganhar) [inline cocos/core/geometry/aabb.ts]: The `warn` invocation will increase 206 bytes of package size.
Probably, we could make it wrap around it with  `DEBUG` macros.

Comment (dumganhar) [reply cocos/core/geometry/aabb.ts]: Or could we put it in `cocos/core/deprecated.ts`?

Comment (minggo) [reply cocos/core/geometry/aabb.ts]: deprecated.ts can not distinguish which function is deprecated.

Comment (minggo) [reply cocos/core/geometry/aabb.ts]: > The `warn` invocation will increase 206 bytes of package size. Probably, we could make it wrap around it with `DEBUG` macros.

Yep. Reasonable, if TS compiler can generate warning message in compiling is better.

**Review by dumganhar — APPROVED** [review]

**Pull request merged**